src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/TestHostHelper.cs

```diff
@@ -67,11 +69,23 @@ internal static class TestHostHelper
                     outputHelper.WriteLine("End Configuration");
 
                     builder.AddInMemoryCollection(configurationValues);
+
+                    builder.AddInMemoryCollection(new Dictionary<string, string>
+                    {
+                        {ConfigurationPath.Combine(ConfigurationKeys.Storage, nameof(StorageOptions.DumpTempFolder)), StorageOptionsDefaults.DumpTempFolder }
+                    });
                 })
-                .ConfigureServices(services =>
+                .ConfigureServices((HostBuilderContext context, IServiceCollection services) =>
                 {
                     services.ConfigureCollectionRules();
                     services.ConfigureEgress();
+
+                    services.AddSingleton<EgressOperationQueue>();
```

Comment:
Do we really need these dependencies? You are egressing directly from what I can see.